tests/run-make/cdylib/rmake.rs

```diff
@@ -0,0 +1,34 @@
+// When the cdylib crate type was added as a variation of dylib, it needed a test to check its
+// function. See <https://github.com/rust-lang/rust/pull/33553>.
```

Comment:
Suggestion: this comment isn't very useful. Maybe we can improve its wording, like:

This test tries to check that basic cdylib libraries can be compiled and linked successfully with C code, that the cdylib itself can depend on another rlib, and that the library can be built with LTO.

- `bar.rs` is a rlib
- `foo.rs` is a cdylib that relies on an extern crate `bar` and defines two `extern "C"` functions:
    - `foo()` which calls `bar::bar()`.
    - `bar()` which implements basic addition.